tutorials/2_Gazebo_and_ROS.ipynb

```diff
@@ -0,0 +1,1330 @@
+    "Right-click on a model now to see what&#39;s available. </p>\n",
+    "\n",
+    "<p><img src='https://bitbucket.org/osrf/gazebo_tutorials/raw/default/guided_b/files/ftu3-mouse-controls.png' width='800px'/></p>\n",
+    "\n",
+    "<h2>Linking Gazebo and ROS</h2>\n",
+    "\n",
+    "<p>To achieve cooperation between with stand-alone Gazebo and ROS, a set of ROS packages named gazebo_ros_pkgs provide wrappers around the stand-alone Gazebo core. They enable the necessary interfaces to simulate a robot in Gazebo using ROS messages and services</p>\n",
+    "\n",
+    "<img src=\"https://bitbucket.org/osrf/gazebo_tutorials/raw/default/ros_overview/figs/775px-Gazebo_ros_api.png\">\n",
+    "<h3 style=\"padding-left:32%\"><u>Overview of Gazebo ROS packages</u></h3>\n",
+    "\n",
+    "<h2>TF Package</h2>\n",
+    "\n",
+    "<p>Before we start writing URDF of Robot, lets quickly go throught the concept of TF. TF is a package that lets the user keep track of multiple coordinate frames over time. tf maintains the relationship between coordinate frames in a tree structure buffered in time, and lets the user transform points, vectors, etc between any two coordinate frames at any desired point in time. You want to see what tf can do instead of just reading about it? Check out the <a href=\"http://wiki.ros.org/tf/Tutorials/Introduction%20to%20tf\">tf introduction demo</a>. A robotic system typically has many 3D coordinate frames that change over time, such as a world frame, base frame, gripper frame, head frame, etc. tf keeps track of all these frames over time, and allows you to ask questions like:</p>\n",
+    "<ul>\n",
+    "<li>Where was the head frame relative to the world frame, 5 seconds ago?</li>\n",
+    "<li>What is the pose of the object in my gripper relative to my base?</li>\n",
+    "<li>What is the current pose of the base frame in the map frame?</li>\n",
+    "</ul>\n",
+    "\n",
+    "<p>You can get started on the introduction to tf tutorial. For a complete list of all tf and tf-related <a href=\"http://wiki.ros.org/tf/Tutorials\">tutorials</a> check out the tutorials page.</p>\n",
+    "\n",
+    "\n",
+    "<h2>Universal Robotic Description Format (URDF)</h2>\n",
+    "<p>The Universal Robotic Description Format (URDF) is an XML file format used in ROS to describe all elements of a robot. To use a URDF file in Gazebo, some additional simulation-specific tags must be added to work properly with Gazebo. This tutorial explains the necessary steps to successfully use your URDF-based robot in Gazebo, saving you from having to create a separate Simulation Description Format (SDF) file from scratch and duplicating description formats. Under the hood, Gazebo will then convert the URDF to SDF automatically. Xacro (XML Macros) is an XML macro language. It is very useful to make shorter and clearer robot descriptions.</p>\n",
+    "\n",
+    "<h2>URDF Tags</h2>\n",
+    "<p>Main tags we will use in writing our URDF are as follows:</p>\n",
```

Comment:
The main tags used in URDF files are:

This section should be a subsection of URDF